lxd/instance/drivers/driver_qemu.go

```diff
@@ -5157,3 +5157,40 @@ func (vm *qemu) writeInstanceData() error {
 
 	return nil
 }
+
+// Info returns "qemu" and the currently loaded qemu version, 0 if not found.
+func (vm *qemu) Info() instance.Info {
+	var qemuVersion, qemuBinary, qemuPath string
+	var out []byte
+	var err error
+	data := instance.Info{
+		Name:    "qemu",
+		Version: "0",
+	}
+
+	if vm.architecture == 0 {
+		vm.architecture, err = osarch.ArchitectureGetLocalID()
+	}
+	if err != nil {
+		return data
+	}
+
+	qemuBinary, _, err = vm.qemuArchConfig()
+	if err != nil {
+		return data
+	}
```

Comment:
I think it'd be cleaner to make `qemuArchConfig` take an architecture as an argument, then pass `d.architecture` from its current call in `Start()` and pass in `osarch.ArchitectureGetLocalID()` in the case of `Info.

Reply:
I can adjust ``qemuArchConfig`` to take an architecture as an argument, but it wouldn't actually clean up the code here very much since ``osarch.ArchitectureGetLocalID()`` returns an error value along with the ID value, it'd just end up being the same thing without the extra if statement to check for an unnecessary call, and using a temporary variable instead of ``d.architecture``.

Reply:
Implemented in most recent commit